src/server/pps/server/master.go

```diff
@@ -421,6 +433,31 @@ func (a *apiServer) monitorPipeline(pachClient *client.APIClient, pipelineInfo *
 	}
 }
 
+func (a *apiServer) monitorCrashingPipeline(ctx context.Context, op *pipelineOp) {
+	defer a.cancelMonitor(op.name)
+For:
+	for {
+		select {
+		case <-time.After(crashingBackoff):
+		case <-ctx.Done():
+			break For
+		}
+		time.Sleep(crashingBackoff)
+		workersUp, err := op.allWorkersUp()
+		if err != nil {
+			log.Printf("error in monitorCrashingPipeline: %v", err)
+			continue
+		}
+		if workersUp {
+			if err := op.setPipelineState(pps.PipelineState_PIPELINE_RUNNING, ""); err != nil {
```

Comment:
Do you think there's any way this codepath could be entered when the pipeline is in `PAUSED`? I know that `step()` cancels any `monitorCrashingPipeline` goros when it detects an update, which might be all that's needed.

If it *could* happen, then `transitionPipelineState` could be used to confirm that the pipeline is still in `CRASHING` when this is called, but I can't tell if it's necessary or not.